code/game/machinery/computer/robot.dm

```diff
@@ -88,21 +91,38 @@
 			if(allowed(usr))
 				var/mob/living/silicon/robot/R = locate(params["ref"]) in GLOB.silicon_mobs
 				if(can_control(usr, R) && !..())
-					if(isAI(usr) && (R.ai_lockdown && R.lockcharge || !R.lockcharge) || !isAI(usr))
-						R.ai_lockdown = FALSE
-						if(isAI(usr) && !R.lockcharge)
+					if(isAI(usr))
+						if(R.ai_lockdown && R.lockcharge)
+							R.ai_lockdown = FALSE
+							src.lock_unlock_borg(R)
+						else if(!R.ai_lockdown && R.lockcharge)
+							to_chat(usr, span_danger("Cyborg locked by an user with superior permissions."))
+						else if(!R.lockcharge)
 							R.ai_lockdown = TRUE
-						message_admins(span_notice("[ADMIN_LOOKUPFLW(usr)] [!R.lockcharge ? "locked down" : "released"] [ADMIN_LOOKUPFLW(R)]!"))
-						log_silicon("[key_name(usr)] [!R.lockcharge ? "locked down" : "released"] [key_name(R)]!")
-						log_combat(usr, R, "[!R.lockcharge ? "locked down" : "released"] cyborg")
-						R.SetLockdown(!R.lockcharge)
-						to_chat(R, !R.lockcharge ? span_notice("Your lockdown has been lifted!") : span_alert("You have been locked down!"))
-						if(R.connected_ai)
-							to_chat(R.connected_ai, "[!R.lockcharge ? span_notice("NOTICE - Cyborg lockdown lifted") : span_alert("ALERT - Cyborg lockdown detected")]: <a href='?src=[REF(R.connected_ai)];track=[html_encode(R.name)]'>[R.name]</a><br>")
+							src.lock_unlock_borg(R)
 					else
-						to_chat(usr, span_danger("Cyborg locked by an user with superior permissions."))
+						if(isnull( locked_down_borg)&& !R.lockcharge) //If there is no borg locked down by the console yet
+							src.lock_unlock_borg(R, src.loc.loc.name)
+							R.ai_lockdown = FALSE //Just in case I'm stupid
+							locked_down_borg = R
+							RegisterSignal(R, COMSIG_PARENT_QDELETING, PROC_REF(borg_destroyed))
+						else if(locked_down_borg == R) //If the borg locked down by the console is the same as the one we're trying to unlock
+							src.lock_unlock_borg(R)
+							UnregisterSignal(R, COMSIG_PARENT_QDELETING)
+							locked_down_borg = null
```

Comment:
The exact code block of "toggle lock, unregister signal, clear reference" is repeated several times, can we make this a proc?